### PR TITLE
chore: update to ava@3

### PR DESCRIPTION
**Pull request by SimenB**
Should in theory clear out the dependabot warnings in the repo

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -90,24 +91,15 @@
...
   "lint-staged": {
-    "*.js": [
-      "eslint --fix",
-      "git add"
-    ],
-    "*.{ts,md,json}": [
-      "prettier --write",
-      "git add"
-    ]
+    "*.js": "eslint --fix",
```
Comment (SimenB) [inline package.json:94]: image

File: package.json [hunk 2]
```diff
@@ -41,13 +41,13 @@
     "examples"
   ],
   "devDependencies": {
+    "@ava/babel": "^1.0.1",
     "@types/ip": "^1.1.0",
     "@types/murmurhash3js": "^3.0.2",
-    "@types/nock": "^11.1.0",
```
Comment (SimenB) [inline package.json:46]: `nock` ships with types

**Comment by SimenB** [comment]
Interestingly, this gave 3 test failures on node 10. Seems to be some sort of race condition. The tests looked somewhat faulty to me either way - e.g. multiple calls to `end`.

I added `t.plan` which made the tests fail with "wrong number of assertions" rather than "test already done", then fixed the tests (I think)

**Review by chriswk — APPROVED** [review]
This looks good. We should probably merge this before looking at my bigger upgrade all dependencies PR:#206 . And then rebase the #206 branch on top of master after merging this, reducing the surface that touches.

**Comment by chriswk** [comment]
I updated #206 to include what you've done here :)

**Review by ivarconr — APPROVED** [review]

**Pull request merged**